packages/playback-core/src/renditions.ts

```diff
@@ -0,0 +1,103 @@
+import Hls from './hls';
+import type { HlsInterface } from './hls';
+import type { MediaTracks } from './types';
+import type { TrackEvent } from 'media-tracks';
+
+export function setupRenditions(customMediaEl: MediaTracks, hls: HlsInterface) {
+  if (!('videoTracks' in customMediaEl)) return;
+
+  // Create a map to save the unique id's we create for each level and rendition.
+  // hls.js uses the levels array index primarily but we'll use the id to have a
+  // 1 to 1 relation from rendition to level.
+  const levelIdMap = new WeakMap();
+
+  hls.on(Hls.Events.MANIFEST_PARSED, function (event, data) {
+    removeAllVideoTracks();
+
+    const videoTrack = customMediaEl.addVideoTrack('main');
+    videoTrack.selected = true;
+
+    for (const [id, level] of data.levels.entries()) {
+      const videoRendition = videoTrack.addRendition(
+        level.url[0],
+        level.width,
+        level.height,
+        level.videoCodec,
+        level.bitrate
+      );
+
+      // The returned levels all have an id of `0`, save the id in a WeakMap.
+      levelIdMap.set(level, `${id}`);
+
+      videoRendition.id = `${id}`;
+      videoRendition.enabled = hls.autoLevelEnabled;
+
+      if (id == data.firstLevel) {
+        videoRendition.enabled = true;
+        videoRendition.active = true;
+      }
+    }
+  });
+
+  // Fired when a level is removed after calling `removeLevel()`
+  hls.on(Hls.Events.LEVELS_UPDATED, function (event, data) {
+    const videoTrack = customMediaEl.videoTracks[0];
+    if (!videoTrack) return;
+
+    const levelIds: string[] = data.levels.map((l) => levelIdMap.get(l));
+
+    for (const rendition of videoTrack.renditions) {
+      if (rendition.id && !levelIds.includes(rendition.id)) {
+        videoTrack.renditions.remove(rendition);
+      }
+    }
+  });
+
+  hls.on(Hls.Events.LEVEL_SWITCHED, function (event, data) {
+    const videoTrack = customMediaEl.videoTracks[0];
+    if (!videoTrack) return;
+
+    // data.level returns an index of the level, not an id.
+    const levelId = levelIdMap.get(hls.levels[data.level]);
+    const rendition = videoTrack?.renditions.getRenditionById(levelId);
+    if (rendition) {
+      rendition.active = true;
```

Comment:
question(blocking): Shouldn't we also be setting any previously active rendition to false here?

Reply:
that's done internally 
https://github.com/muxinc/media-tracks/blob/2cf67e680301a23100a17c669ad50ac8989f6771/src/video-rendition.ts#L56-L58